packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      const user1CreateAllowedSet = `
+        ${privateResultSet}
+        ownerContent
+      `;
+      const createResult1 = await user1TodoHelper.create(
+        createResultSetName,
+        { ...todoPrivateFields, ownerContent: 'Owner Content' },
+        user1CreateAllowedSet,
+      );
+      expect(createResult1.data[createResultSetName].id).toBeDefined();
+      expect(createResult1.data[createResultSetName].owner).toEqual(userName1);
+      expect(createResult1.data[createResultSetName].authors).toEqual([userName1]);
+      todoPrivateFields['id'] = createResult1.data[createResultSetName].id;
+      // protected fields are nullified in mutation responses
+      expectNullFields(createResult1.data[createResultSetName], ['privateContent', 'ownerContent']);
+
+      const privateAndPublicSet = `
+        ${privateResultSet}
+        publicContent
+      `;
+      // cannot update a record with public protected field
+      await expect(
+        async () =>
+          await user1TodoHelper.update(updateResultSetName, { ...todoPrivateFields, publicContent: 'Public Content' }, privateAndPublicSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      const privateAndOwnerSet = `
+        ${privateResultSet}
+        ownerContent
+      `;
+      // cannot update a record with owner protected field that does not allow update operation
+      await expect(
+        async () =>
+          await user1TodoHelper.update(updateResultSetName, { ...todoPrivateFields, ownerContent: 'Owner Content' }, privateAndOwnerSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      /*
+      const privateAndOwnersSet = `
+        ${privateResultSet}
+        ownersContent
+      `;
+      // non-owner cannot update a record with dynamic owner list protected field
+      await expect(async () => await user2TodoHelper.update(updateResultSetName, { ...todoPrivateFields, ownersContent: 'Owners Content'}, privateAndOwnersSet)).rejects.toThrowErrorMatchingInlineSnapshot(
+        expectedOperationError(updateResultSetName, 'Mutation'),
+      );
+      */
+
+      // cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ id: todoPrivateFields['id'] }, privateAndPublicSet, false, 'all');
+      checkOperationResult(
+        getResult1,
+        { ...todoPrivateFields, publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, privateAndPublicSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todoPrivateFields['id'], true);
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
```

Comment:
Here and throughout: How are we tracking the need to re-enable these cases? This test is really long, and it would be easy to miss a commented out snippet like this.

For cases we need to re-enable after the utility work is complete, consider adding an easily searchable todo slug in the comment, or add to our backlog ticket a link to this code line.

Reply:
Added todo slugs for easy tracking.